addons/stock/models/stock_picking.py

```diff
@@ -797,10 +797,12 @@ def _search_delay_alert_date(self, operator, value):
     def _onchange_picking_type(self):
         if self.picking_type_id and self.state == 'draft':
             self = self.with_company(self.company_id)
-            (self.move_ids | self.move_ids_without_package).update({
+            moves_update_picking_type = (self.move_ids | self.move_ids_without_package).filtered(lambda m: m.picking_type_id != self.picking_type_id)
+            moves_update_picking_type.update({
                 "picking_type_id": self.picking_type_id,  # The compute store doesn't work in case of One2many inverse (move_ids_without_package)
                 "company_id": self.company_id,
             })
+            ((self.move_ids | self.move_ids_without_package) - moves_update_picking_type).company_id = self.company_id
```

Comment:
cleaner and shorter like this
```suggestion
            # The compute store doesn't work in case of One2many inverse
            (self.move_ids | self.move_ids_without_package).filtered(lambda m: m.picking_type_id != self.picking_type_id).picking_type_id = self.picking_type_id
            (self.move_ids | self.move_ids_without_package).company_id = self.company_id
```

Reply:
Agreed, but with +1 line